src/resolve/resolved-dns-query.c

```diff
@@ -1377,3 +1382,12 @@ bool dns_query_fully_authoritative(DnsQuery *q) {
          * explicitly check previous redirects here.) */
         return (q->answer_query_flags & SD_RESOLVED_FROM_MASK & ~(SD_RESOLVED_FROM_TRUST_ANCHOR | SD_RESOLVED_FROM_ZONE)) == 0;
 }
+
+bool dns_question_has_record_types(DnsQuestion *q, Set *types) {
```

Comment:
I do not think this should be exported. Please make it static and move just before the declaration of `dns_query_new()`.

Reply:
removed export